app/org/sagebionetworks/bridge/dynamodb/DynamoUploadDao.java

```diff
@@ -148,9 +148,7 @@ public Upload getUpload(@Nonnull String uploadId) {
 
         String nextPageOffsetKey = (page.getLastEvaluatedKey() != null) ? page.getLastEvaluatedKey().get(UPLOAD_ID).getS() : null;
 
-        int total = mapper.count(DynamoUpload2.class, createCountQuery(studyId.getIdentifier(), startTime, endTime));
-
-        return new PagedResourceList<>(uploadList, null, pageSize, total)
+        return new PagedResourceList<>(uploadList, null, pageSize, uploadList.size())
```

Comment:
We discussed offline and this code does not lie in the path of the call that is used by the UI and that currently implements pagination off a count of all the records in the query... OTOW this is fine to merge from my perspective.